dxp/demystifying_aui_script_and_liferay-frontend_component_tags.md

```diff
@@ -0,0 +1,114 @@
+# Demystifying `aui:script` and `liferay-frontend:component` tags
+
+On JSP context you may notice a huge number of `aui:script` and `liferay-frontend:component` tags.
+
+## `aui:script` tag
+
+### What it does:
+
+This taglib allow users to use and require another JS module/resources inside a JSP.
+
+This taglib have three possible use cases:
+
+1. Pure JS on the JSP
+2. Import a AUI/YUI module
+3. Import a ES6 module
+
+#### Pure JS on the JSP
+
+By just placing JavaScript code inside an `aui:script` tag, you can take advantage of optimizing scripts and place them to be made available in the final compressed JavaScript bundle, see [combo for more details](https: / /github.com/liferay/liferay-frontend-guidelines/blob/45bad0334cb467e45e68aee322a2a4bc0b6979d3/dxp/javaScript_minification.md#comboservlet). It is worth mentioning that the same rule applies for the following use cases.
+
+#### Import an AUI/YUI module using `use` property
+
+When using the `use` property, the AUI / YUI module listed in the` use` list will be imported to the script data. If you want to know a little more about ScriptData, see this [link on how resources are injected into DXP](https://github.com/liferay/liferay-frontend-guidelines/blob/ea185686db8d562ac11cc2b95d2e9e7d7a0c6547/dxp/resource_injection.md#scriptdata).
```

Comment:
I might be wrong about this, but it doesn't match my mental model of how things work. AUI modules are loaded asynchronously and not "imported to the script data" (that makes it sound like they are made available when the page loads).

Reply:
Yep, seems a bit lost. I'm working on tracking better as https://github.com/liferay/liferay-frontend-guidelines/pull/226#discussion_r499970785 comment :)

Reply:
Yes, considering this to be injected via scriptData is completely wrong. AUI/YUI modules are loaded asynchronously. 

Replaced with:

_Each AUI / YUI module listed as a value for the `use` property will be [loaded asyncronously by the `AUI().use` function](https://github.com/yui/yui3/blob/master/src/yui/js/yui.js#L1323..L1331)._